### ISSUE TITLE
Twitter not working

### ISSUE THREAD

**Issue by djgd** [pavelk2/social-feed#268]
There appears to be an issue with twitter, no tweets are being displayed for any twitter account

**Comment by bluelevel-media** [comment]
Yes I am having the same issue across all my sites using this. Looks like the social-feed demo is also doing the same... hoping that twitter hasn't been changing stuff again!

**Comment by bluelevel-media** [comment]
wasn't getting any console errors before but not suddenly I am getting thousands (literally) of console errors building up with the same errors:

XMLHttpRequest cannot load https://api.jublo.net/codebird/oauth2/token. Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://www.XXXX.co.uk' is therefore not allowed access. The response had HTTP status code 502.

**Comment by HartLarsson** [comment]
i've see that too, maybe codebird have changed something about token/access/security in API
we need to investigate

**Comment by RodCoxi** [comment]
Hey, exactly the same for me ! I'm a beginer but i've seen that the tweeter api's link : http://api.tweecool.com/ doesn't exist ... it's in 1&1

**Comment by mutiullah7** [comment]
Hey, same for me but im getting a console error
POST **https://api.jublo.net/codebird/oauth2/token 500 (Internal Server Error)**  **codebird.js?ver=4.8:525**

**Comment by dylanda** [comment]
Same issue here... That might help or be related : https://github.com/jublonet/codebird-cors-proxy/issues/8

**Comment by djgd** [comment]
More likely this one https://github.com/jublonet/codebird-js/issues/135

**Comment by pferder** [comment]
I'm getting Twitter posts but takes some time, 1 minute or 2

**Comment by Morphinof** [comment]
Same here, impossible to load tweets, to prevent the plugin from requesting https://api.jublo.net/codebird/oauth2/token too much i've done this : 
`$.ajax({
                    url: 'https://api.jublo.net/codebird/oauth2/token',
                    timeout: 1000,
                    success: function () { $('.social-feed-container').socialfeed(options); },
                    error: function () {
// Removing twitter config
                        delete options.twitter;
                        $('.social-feed-container').socialfeed(options);
                    }
                });`

Any news around the issue ?

**Comment by mutiullah7** [comment]
@HartLarsson @pferder Same here, im getting tweets now but after 2 or 3 minutes.

**Comment by Dazdd** [comment]
@mutiullah7 Same here, I disabled the twitter function and everything else is working perfectly. Awaiting an update to fix the twitter issue.

**Comment by djgd** [comment]
Disabled, looking for an alternative - codebird have closed the issue seemingly without a resolution

**Comment by HartLarsson** [comment]
i still following the thread here: https://github.com/jublonet/codebird-js/issues/137
seems is a platform codebird issue, i see that after a minute or so the stream comes from twitter without any error in console.
Is like the jublonet server is overloaded and keep requests in queue

**Comment by mutiullah7** [comment]
@HartLarsson any ideas when it will be fixed?

**Comment by HartLarsson** [comment]
i've no idea because depends on an external library/service. If you read the codebird issue tracker there are many people with problems too.

follow: https://github.com/jublonet/codebird-js/issues/135

**Comment by joshuaatkins** [comment]
Hi guys,

We (Jublo - Codebird) have made (considerable) changes to our infrastructure to continue our service for you. 
Can you confirm if this has made the improvements required?

Thanks,

Josh

**Comment by Morphinof** [comment]
Everything is working fine again, thanks !

**Comment by HartLarsson** [comment]
Hi @joshuaatkins all workings ok right now thank you for your prompt replay!

**Comment by Hypocrite** [comment]
We are still experiencing issues. Do we have to update Codebird to fix the issue?

**Comment by martijngastkemper** [comment]
You have to update Codebird. We don't manage that server.

**Comment by Hypocrite** [comment]
I downloaded the latest Codebird version from https://github.com/jublonet/codebird-js but it seems it was last updated a year ago. 

It does not give a response from https://api.jublo.net/codebird/oauth2/token. I will post a comment to the Codebird issue.

**Comment by harman993** [comment]
is there any updates from codebird ...im also getting 502 error while accessing twitter

**Comment by CrySensai** [comment]
same here again  "No 'Access-Control-Allow-Origin"

**Comment by mynetx** [comment]
@all Hello there, the issues should be resolved now, can you test?

**Comment by HartLarsson** [comment]
now we have the same problem again @jublo

**Comment by mynetx** [comment]
Can you test again? We made more changes.

**Comment by mutiullah7** [comment]
@jublo Twitter feed again stopped working
`OPTIONS https://api.jublo.net/codebird/oauth2/token net::ERR_CONNECTION_TIMED_OUT`

**Comment by pferder** [comment]
Same here! 
`codebird.js:525 OPTIONS https://api.jublo.net/codebird/oauth2/token net::ERR_CONNECTION_TIMED_OUT`

**Comment by jean-emmanuel** [comment]
It's a php bug on Jublo's proxy side. I solved it by installing https://github.com/jublonet/codebird-cors-proxy directly on my server. You have to set twitter's proxy option as follow : 
```
twitter:{
    proxy:"https://url/to/codebird-cors-proxy/",
    accounts: //etc
}
```

**Comment by mynetx** [comment]
@jean-emmanuel You meant code like this?

```js
cb.setProxy("https://url/to/codebird-cors-proxy/");
```

**Comment by jean-emmanuel** [comment]
I was refering to jquery.socialfeed's configuration, but yes that's what gets called in the end (https://github.com/pavelk2/social-feed/blob/gh-pages/js/jquery.socialfeed.js#L231).

**Comment by marcustisater** [comment]
@jublo any updates? Still getting 500

**Comment by joshuaatkins** [comment]
@marcustisater are you receiving 500 now? Or was it from around 20:00-22:00 GMT yesterday?

**Comment by marcustisater** [comment]
Hello @joshuaatkins, thanks for getting in touch. I am receiving them now as well. But yes, it worked yesterday.

See https://api.jublo.net/codebird/1.1/statuses/user_timeline.json?id=X

**Comment by mutiullah7** [comment]
@marcustisater im too getting Bad Authentication error
https://api.jublo.net/codebird/1.1/statuses/user_timeline.json?id=X

**Comment by joshuaatkins** [comment]
Can you provide a sample of the code you are using? I am successfully able to query https://api.jublo.net/codebird/1.1/statuses/user_timeline.json

**Comment by mutiullah7** [comment]
Just checked, its back now, working fine now

**Comment by mutiullah7** [comment]
@joshuaatkins Just checked, it's working now

**Comment by marcustisater** [comment]
Working, might be DNS. Thanks @joshuaatkins

**Comment by iKomettech** [comment]
https://api.jublo.net/codebird/1.1/statuses/user_timeline.json?id=xxx&count=4&tweet_mode=extended 

404 Error in Developer Console but Twitter post displayed.Any solution for this issue
